cit_041a_gtfs_point_locations/contents/src/__init__.py

```diff
@@ -41,130 +49,187 @@
+    '''
+    Given a Resource Watch dataset's API ID and a datetime,
+    this function will update the dataset's 'last update date' on the API with the given datetime
+    INPUT   dataset: Resource Watch API dataset ID (string)
+            date: date to set as the 'last update date' for the input dataset (datetime)
+    '''
+    # generate the API url for this dataset
+    apiUrl = f'http://api.resourcewatch.org/v1/dataset/{dataset}'
+    # create headers to send with the request to update the 'last update date'
     headers = {
-        'Content-Type': 'application/json',
-        'Authorization': os.getenv('apiToken')
+    'Content-Type': 'application/json',
+    'Authorization': os.getenv('apiToken')
     }
+    # create the json data to send in the request
     body = {
-        "dataLastUpdated": date.isoformat()
+        "dataLastUpdated": date.isoformat() # date should be a string in the format 'YYYY-MM-DDTHH:MM:SS'
     }
+    # send the request
     try:
-        r = requests.patch(url=apiUrl, json=body, headers=headers)
-        logging.info('[lastUpdated]: SUCCESS, ' + date.isoformat() + ' status code ' + str(r.status_code))
+        r = requests.patch(url = apiUrl, json = body, headers = headers)
+        logging.info('[lastUpdated]: SUCCESS, '+ date.isoformat() +' status code '+str(r.status_code))
         return 0
     except Exception as e:
-        logging.info('[lastUpdated]: ' + str(e))
+        logging.error('[lastUpdated]: '+str(e))
 
+'''
+FUNCTIONS FOR THIS DATASET
 
-def get_most_recent_date(table):
-    # r = cartosql.getFields(TIME_FIELD, table, f='csv', post=True)
-    r = getFields(TIME_FIELD, table, f='csv', post=True)
-    dates = r.text.split('\r\n')[1:-1]
-    dates.sort()
-    most_recent_date = datetime.datetime.strptime(dates[-1], '%Y-%m-%d %H:%M:%S')
-    return most_recent_date
-
-
-def formatDate(date):
-    """ Parse input date string and write in output date format """
-    return datetime.datetime.strptime(date, INPUT_DATE_FORMAT) \
-        .strftime(DATE_FORMAT)
-
-
-def getFilename(date):
-    '''get filename from datestamp CHECK FILE TYPE'''
-    return os.path.join(DATA_DIR, '{}.csv'.format(
-        FILENAME.format(date=date.strftime('%Y%m%d'))))
-
+The functions below have been tailored to this specific dataset.
+They should all be checked because their format likely will need to be changed.
+'''
 
 def getGeom(lon, lat):
-    '''Define point geometry from latitude and longitude'''
+    '''
+    Define point geometry from latitude and longitude
+    '''
+    # construct geojson using values from lon, lat columns
     geometry = {
         'type': 'Point',
         'coordinates': [float(lon), float(lat)]
     }
     return geometry
 
-
 def convert_time_since_epoch(timestamp):
-    '''Function to convert seconds since the epoch to human readable time'''
+    '''
+    Function to convert seconds since the epoch to human readable time
```

Comment:
```suggestion
    Function to convert seconds since the epoch to human-readable time
    INPUT   timestamp: date, formatted as time since linux epoch (integer)
    RETURN  formatted string of date (string)
```